File: coupon/coupon-api/src/main/java/dev/be/coupon/api/coupon/presentation/dto/CouponIssueResponse.java
package dev.be.coupon.api.coupon.presentation.dto;

import dev.be.coupon.api.coupon.application.dto.CouponIssueResult;

import java.time.LocalDateTime;
import java.util.UUID;

public record CouponIssueResponse(
        UUID userId,
        UUID couponId,
        boolean used,
        LocalDateTime issuedAt
) {
    public static CouponIssueResponse from(final CouponIssueResult result) {
        return new CouponIssueResponse(
                result.userId(),
                result.couponId(),
                result.used(),
                result.issuedAt()
        );
    }
}

File: coupon/coupon-api/src/main/kotlin/dev/be/coupon/api/coupon/presentation/dto/CouponIssueResponse.kt
package dev.be.coupon.api.coupon.presentation.dto

import dev.be.coupon.api.coupon.application.dto.CouponIssueResult
import java.time.LocalDateTime
import java.util.*

data class CouponIssueResponse(
    val userId: UUID,
    val couponId: UUID,
    val used: Boolean,
    val issuedAt: LocalDateTime
) {
    companion object {
        fun from(result: CouponIssueResult): CouponIssueResponse {
            return CouponIssueResponse(
                userId = result.userId,
                couponId = result.couponId,
                used = result.used,
                issuedAt = result.issuedAt
            )
        }
    }
}


